src/west/app/main.py

```diff
@@ -427,6 +561,42 @@ def run_command(self, argv):
         except WestNotFound as wnf:
             self.cmd.die(str(wnf))
 
+    def handle_early_arg_errors(self, early_args):
+        # If early_args indicates we should error out, handle it
+        # gracefully. This provides more user-friendly output than
+        # argparse can do on its own.
```

Comment:
It's more user-friendly... unless the manifest fails to import. Then the original, relevant error message is lost, see issue #726 and tentative mitigation #727